File: App/Services/Microservices.Demo.Product.Service/Domain/Products/Entities/ChoiceQuestion.cs
namespace Microservices.Demo.Products.Service.Domain.Products.Entities
{
    public class ChoiceQuestion : Question
    {
        public ChoiceQuestion()
        {
        }

        public ChoiceQuestion(string code, int index, string text, List<Choice> choices) : base(code, index, text)
        {
            Choices = choices;
        }

        public List<Choice> Choices { get; set; }

        public static List<Choice> YesNoChoice()
        {
            return new List<Choice>
        {
            new("YES", "Yes"),
            new("NO", "No")
        };
        }
    }

}

File: App/Clients/ng-microservices-demo-client/src/app/domain/products/entities/choice-question.ts
import { AutoMap } from '@automapper/classes';
import { Choice } from './choice';
import { Question } from './question';

export class ChoiceQuestion extends Question {
  @AutoMap(() => [Choice])
  public choices: Choice[];

  constructor(
    questionCode: string,
    index: number,
    text: string,
    choices: Choice[]
  ) {
    super(questionCode, index, text);
    this.choices = choices;
  }
}
